src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EmailAddressAttribute.cs

```diff
@@ -32,7 +25,10 @@ public override bool IsValid(object value)
             }
 
             var valueAsString = value as string;
-            return valueAsString != null && _regex.Match(valueAsString).Length > 0;
+            return (valueAsString != null
+                && valueAsString.Count(c => c == '@') == 1
+                && valueAsString[0] != '@'
+                && valueAsString[valueAsString.Length-1] != '@');
```

Comment:
Sure. I haven't come across any worries of that nature in the past but I'm happy to avoid it if you think there's an issue.